### ISSUE TITLE
Provide an extension point to disable IDE0051 on selected members

### ISSUE THREAD

**Issue by AndreasHeisel** [dotnet/roslyn#34986]
We have private methods in our codebase that trigger an IDE0051, because there are no references visible to the analyzer.

These methods get called by reflection and therefore shouldn't be removed. They are all decorated with a custom attribute. 

I found #31036 wich adds a list of attributes that disable IDE0051 on selected members.

My suggestion is to make these list extensible bei either an attribute or a marker interface to decorate custom attributes. Something like

```c#
[
    IndicatesUsage
]
public sealed class MyAttribute : Attribute
{}
```
or

```c#
public sealed class MyAttribute : Attribute, IUsageIndicator
{}
```
Members decorated with such an attribute should not trigger IDE0051:

```c#
[
    MyAttribute
]
private void GetsMagicallyCalled()
{}
```
This would enable library authors to prevent false positives for users of their libraries.

**Version Used**: 
VS2019

**Comment by sharwell** [comment]
Duplicate of #20242

**Comment by AndreasHeisel** [comment]
@sharwell I would say not a exact duplicate, but deals with the same problem.

I think #20242 would fix our problem too. As it was moved back to backlog: Any idea if and when we will see this in VisualStudio?
